### PR TITLE
Complete getByCategory function (Step 2)

### PR DESCRIPTION
**Pull request by smalldoa**
Plz accept

### CODE REVIEW

File: index.html
Comment (VincenzoMeschi) [inline index.html:1]: No need to update index. Commit without the newline

File: src/data-controller.ts [hunk 1]
```diff
@@ -11,14 +11,18 @@ function generateProductHTML(product: Product): string {
 
 function renderProducts(prods: Product[]): void {
     const container = document.getElementById("main-container");
+    if (container) container.innerHTML = "";
```
Comment (VincenzoMeschi) [inline src/data-controller.ts]: This is repetitive with line 16. See below comment

File: src/data-controller.ts [hunk 2]
```diff
@@ -11,14 +11,18 @@ function generateProductHTML(product: Product): string {
...
     prods.map((product) => {
         if (!container) return;
```
Comment (VincenzoMeschi) [inline src/data-controller.ts]: Merge this condition check with the above check. There is no point in checking if `container` exists on each iteration. Add the following code to line 14:

`container ? container.innterHTML = "" : return`

**Review by VincenzoMeschi — CHANGES_REQUESTED** [review]
Minor changes. Good job.

File: src/data-controller.ts (hunk 2 above)
Comment (smalldoa) [reply src/data-controller.ts]: Fixed. Thank you for the suggestion.

File: src/data-controller.ts (hunk 1 above)
Comment (smalldoa) [reply src/data-controller.ts]: Fixed.

**Review by VincenzoMeschi — APPROVED** [review]

**Pull request merged**